toc.md

```diff
@@ -714,30 +880,42 @@ module Workerholic
 end
 ```
 
-In our second iteration, we decided to use a hash instead. Same as before, using the beginning of the day timestamps as hash fields, and this way we push the computation logic down to the Redis level when we retrieve our data. Redis also has a very nice convenient hash method for incrementing fields.
+By doing the above we avoid concurrency issues by relying on Redis, which uses a write lock when executing the `HINCRBY` command.
+Once we have figured out how we wanted to store the data, we need to think about how many data points to store?
 
 ##### How much data?
-Once we have figured out how we wanted to store the data, we need to think about how many data points to store? The goal for us here is to be able to store data for up to a year.
+
+We decided to store data for up to 365 days.
+
+As a starting point we decided to look at how much space would a hash field, with a timestamp as a field key and a count of jobs as a value field, take. To this effect we wrote a short Ruby script that would set 10,000 hash fields in Redis. Then we use `redis-cli` and the `debug object` command in order to get the size of the whole hash.
```

Comment:
As a starting point, we decided to look at how much space would a hash field take in memory, with a timestamp as a field key and a count of jobs as a value field. To this effect we wrote a short Ruby script that would set 10,000 hash fields in Redis. Then we use `redis-cli` and the `debug object` command to get the size of the whole hash.